### ISSUE TITLE
sampling an empty array crashes

### ISSUE THREAD

**Issue by Nodnarb12500** [Crowlang/crow#62]
```
defvar emptyArray (make-array 0)
(sample emptyArray)
```

hangs and very quickly takes all your ram.

**Comment by Bravotic** [comment]
Hah, I never got around to fixing that I guess, I know at one point I was going to...

**Comment by Nodnarb12500** [comment]
i relooked at my empty array test and found something else to be the cause
```
(defvar emptyArray (make-array 0)
(println (sample emptyArray))
```

i forgot a ')' on the defvar line

sample crashes with:
fish: Job 1, '~/Documents/GIT/crow/crow sampl…' terminated by signal SIGFPE (Floating point exception)

**Comment by Bravotic** [comment]
Yeah its still broken though... I have a fix coming later tonight.
